x-pack/metricbeat/module/gcp/metrics/metricset.go

```diff
@@ -352,3 +361,43 @@ func addHostFields(groupedEvents []KeyValuePoint) mapstr.M {
 	}
 	return hostRootFields
 }
+
+func (m *MetricSet) setOrgAndProjectDetails(ctx context.Context) error {
+
+	// Initialize the Cloud Resource Manager service
+	srv, err := cloudresourcemanager.NewService(ctx, m.config.opt...)
+	if err != nil {
+		return fmt.Errorf("failed to create cloudresourcemanager service: %w", err)
+	}
+	// Get Project name
+	project, err := srv.Projects.Get(m.config.ProjectID).Context(ctx).Do()
+	if err != nil {
+		return fmt.Errorf("failed to get project name: %w", err)
+	}
+	if project != nil {
+		m.config.projectName = project.Name
+	}
+	// Get the project ancestor details
+	ancestryResponse, err := srv.Projects.GetAncestry(m.config.ProjectID, &cloudresourcemanager.GetAncestryRequest{}).Context(ctx).Do()
+	if err != nil {
+		return fmt.Errorf("failed to get project ancestors: %w", err)
+	}
+	if len(ancestryResponse.Ancestor) == 0 {
+		return fmt.Errorf("no ancestors found for project '%s'", m.config.ProjectID)
+	}
+	ancestor := ancestryResponse.Ancestor[len(ancestryResponse.Ancestor)-1]
+
+	if ancestor.ResourceId.Type == "organization" {
+		m.config.organizationID = ancestor.ResourceId.Id
+		orgReq := srv.Organizations.Get(fmt.Sprintf("organizations/%s", m.config.organizationID))
+
+		orgDetails, err := orgReq.Context(ctx).Do()
+		if err != nil {
+			return fmt.Errorf("failed to get organization details: %w", err)
+		}
+
+		m.config.organizationName = orgDetails.DisplayName
+	}
+
+	return nil
+}
```

Comment:
@Linu-Elias WDYT about something like this to separate the API calls into two separate methods and add a check for `ancestor.Type`? It feels cleaner to me.

```suggestion
// setOrgAndProjectDetails retrieves and sets the organization and project details.
// This function does not stop execution if a project name or organization details are not found.
func (m *MetricSet) setOrgAndProjectDetails(ctx context.Context) error {
	srv, err := cloudresourcemanager.NewService(ctx, m.config.opt...)
	if err != nil {
		return fmt.Errorf("failed to create cloudresourcemanager service: %w", err)
	}

	// Get project details
	if err := m.setProjectDetails(ctx, srv); err != nil {
		return err
	}

	// Get project ancestry and set organization details
	if err := m.setOrganizationDetails(ctx, srv); err != nil {
		return err
	}

	return nil
}

// setProjectDetails retrieves and sets the project name.
func (m *MetricSet) setProjectDetails(ctx context.Context, srv *cloudresourcemanager.Service) error {
	project, err := srv.Projects.Get(m.config.ProjectID).Context(ctx).Do()
	if err != nil {
		return fmt.Errorf("failed to get project details: %w", err)
	}
	
	m.config.projectName = project.Name
	return nil
}

// setOrganizationDetails retrieves and sets the organization ID and name.
func (m *MetricSet) setOrganizationDetails(ctx context.Context, srv *cloudresourcemanager.Service) error {
	ancestryResponse, err := srv.Projects.GetAncestry(m.config.ProjectID, &cloudresourcemanager.GetAncestryRequest{}).Context(ctx).Do()
	if err != nil {
		return fmt.Errorf("failed to get project ancestors: %w", err)
	}

	if len(ancestryResponse.Ancestor) == 0 {
		return fmt.Errorf("no ancestors found for project '%s'", m.config.ProjectID)
	}

	ancestor := ancestryResponse.Ancestor[len(ancestryResponse.Ancestor)-1]
	if ancestor.ResourceId.Type != "organization" {
		return fmt.Errorf("ancestor is not an organization")
	}

	m.config.organizationID = ancestor.ResourceId.Id
	orgReq := srv.Organizations.Get(fmt.Sprintf("organizations/%s", m.config.organizationID))
	orgDetails, err := orgReq.Context(ctx).Do()
	if err != nil {
		return fmt.Errorf("failed to get organization details: %w", err)
	}

	m.config.organizationName = orgDetails.DisplayName
	return nil
}
```

Reply:
Also should we only log if the call to get project details fails? For example, it might fail due to missing permissions, causing `cloud.project.name` to be empty when it wasn't before. What if we stop execution if the error is due to missing permissions and log other kind of errors? @zmoog

Reply:
++ Yeah, it's cleaner and more readable.

Reply:
yes I have made the necessary changes, but should we stop the execution and return error if it is due to missing permissions and log other kind of errors?

Reply:
> For example, it might fail due to missing permissions, causing cloud.project.name to be empty when it wasn't before. What if we stop execution if the error is due to missing permissions and log other kind of errors? 

@Linu-Elias, what fields will we present if the service account lacks permission to get account and project IDs and names?

Reply:
we already have project ID, if we get any error while fetching project name, organization ID and organization name, we are now only logging the error and continuing with the flow

Reply:
so always present project ID

Reply:
I prefer to document the breaking change carefully and not stop the collection if we can't fetch the org ID and name. 

The field names change, but we continue to deliver the same information (project ID) under a different name. The change is due to the ECS semantics alignment, which is the core of this PR. It is crucial to inform the users of the breaking change.

But this is my take. Can we reach out to PM to double-check?

This conversation reminds me that we also must update the docs to explain to users what permissions they need to add to the service account to get the org ID and name.

Reply:
One more thing I noticed is that if the project call fails the function returns and we don't make the organization call. Since it's acceptable for either of these calls to fail without stopping execution maybe we can join the errors?

```go
	var errs []error
	
	// Get project details
	if err := m.setProjectDetails(ctx, srv); err != nil {
		errs = append(errs, err)
	}
	
	// Get project ancestry and set organization details
	if err := m.setOrganizationDetails(ctx, srv); err != nil {
		errs = append(errs, err)
	}

	return errors.Join(errs...)
```

And for permissions I believe we need:
- `resourcemanager.projects.get`
- `resourcemanager.organizations.get`